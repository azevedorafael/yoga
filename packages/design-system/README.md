# `@gympass/design-system`

> TODO: description

## Usage

```
const designSystem = require('@gympass/design-system');

// TODO: DEMONSTRATE API
```
