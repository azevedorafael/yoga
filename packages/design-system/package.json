{
  "name": "@gympass/design-system",
  "version": "0.0.0",
  "description": "Gympass component library",
  "main": "./src/index.js",
  "keywords": [
    "Gympass",
    "components",
    "styled-components",
    "react",
    "design-system"
  ],
  "homepage": "https://github.com/Gympass/design-system#readme",
  "license": "MIT",
  "publishConfig": {
    "access": "public"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Gympass/design-system.git"
  },
  "bugs": {
    "url": "https://github.com/Gympass/design-system/issues"
  },
  "dependencies": {
    "@gympass/common": "*",
    "@gympass/tokens": "*",
    "prop-types": "^15.7.2"
  },
  "devDependencies": {
    "@testing-library/react": "^9.1.4",
    "@testing-library/react-native": "^4.1.0",
    "react": "^16.9.0",
    "react-dom": "^16.9.0"
  },
  "peerDependencies": {
    "react": "^16.9.0",
    "react-dom": "^16.9.0",
    "styled-components": "^4.4.0"
  }
}
