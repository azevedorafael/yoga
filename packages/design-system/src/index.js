import ThemeProvider, { themes } from './ThemeProvider';
import Button from './Button';
import List from './List';

export { ThemeProvider, themes, Button, List };
