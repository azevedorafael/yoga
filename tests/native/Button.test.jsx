import React from 'react';
import { render, toJSON } from '@testing-library/react-native';
import { ThemeProvider, Button } from '@gympass/design-system';

describe('<Button />', () => {
  describe('Snapshots', () => {
    describe('Without props', () => {
      it('should match snapshot with default Button', () => {
        const { container } = render(
          <ThemeProvider>
            <Button>Foo</Button>
          </ThemeProvider>,
        );
        expect(toJSON(container)).toMatchSnapshot();
      });
    });
  });
});
