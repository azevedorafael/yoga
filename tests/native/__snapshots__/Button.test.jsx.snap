// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Snapshots With disabled prop should match snapshot with default Button 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableWithoutFeedback
    disabled={true}
  >
    <View
      disabled={true}
      full={false}
      inverted={false}
      pressed={false}
      small={false}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "backgroundColor": "#DBDBE8",
            "borderColor": "#DBDBE8",
            "borderRadius": 9999,
            "borderStyle": "solid",
            "borderWidth": 2,
            "height": 48,
            "justifyContent": "center",
            "paddingLeft": 20,
            "paddingRight": 20,
          },
        ]
      }
    >
      <Text
        disabled={true}
        inverted={false}
        pressed={false}
        style={
          Array [
            Object {
              "color": "#9898A6",
              "fontSize": 14,
              "fontWeight": "700",
              "textAlign": "center",
            },
          ]
        }
      >
        Gympass
      </Text>
    </View>
  </TouchableWithoutFeedback>
</View>
`;

exports[`<Button /> Snapshots With disabled prop should match snapshot with outline Button 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableWithoutFeedback
    disabled={true}
  >
    <View
      disabled={true}
      full={false}
      inverted={false}
      pressed={false}
      small={false}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "backgroundColor": "transparent",
            "borderColor": "#9898A6",
            "borderRadius": 9999,
            "borderStyle": "solid",
            "borderWidth": 2,
            "height": 48,
            "justifyContent": "center",
            "paddingLeft": 20,
            "paddingRight": 20,
          },
        ]
      }
    >
      <Text
        disabled={true}
        inverted={false}
        pressed={false}
        style={
          Array [
            Object {
              "color": "#9898A6",
              "fontSize": 14,
              "fontWeight": "700",
              "textAlign": "center",
            },
          ]
        }
      >
        Gympass
      </Text>
    </View>
  </TouchableWithoutFeedback>
</View>
`;

exports[`<Button /> Snapshots With disabled prop should match snapshot with text Button 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableWithoutFeedback
    disabled={true}
  >
    <View
      disabled={true}
      full={false}
      pressed={false}
      small={false}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "backgroundColor": "transparent",
            "borderColor": "transparent",
            "borderRadius": 9999,
            "borderStyle": "solid",
            "borderWidth": 2,
            "height": 48,
            "justifyContent": "center",
            "paddingLeft": 20,
            "paddingRight": 20,
          },
        ]
      }
    >
      <Text
        disabled={true}
        pressed={false}
        style={
          Array [
            Object {
              "color": "#9898A6",
              "fontSize": 14,
              "fontWeight": "700",
              "textAlign": "center",
            },
          ]
        }
      >
        Gympass
      </Text>
    </View>
  </TouchableWithoutFeedback>
</View>
`;

exports[`<Button /> Snapshots With full prop should match snapshot with default Button 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableWithoutFeedback
    full={true}
  >
    <View
      disabled={false}
      full={true}
      inverted={false}
      pressed={false}
      small={false}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "backgroundColor": "#F46152",
            "borderColor": "#F46152",
            "borderRadius": 9999,
            "borderStyle": "solid",
            "borderWidth": 2,
            "height": 48,
            "justifyContent": "center",
            "paddingLeft": 20,
            "paddingRight": 20,
            "width": "100%",
          },
        ]
      }
    >
      <Text
        disabled={false}
        inverted={false}
        pressed={false}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 14,
              "fontWeight": "700",
              "textAlign": "center",
            },
          ]
        }
      >
        Gympass
      </Text>
    </View>
  </TouchableWithoutFeedback>
</View>
`;

exports[`<Button /> Snapshots With full prop should match snapshot with outline Button 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableWithoutFeedback
    full={true}
  >
    <View
      disabled={false}
      full={true}
      inverted={false}
      pressed={false}
      small={false}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "backgroundColor": "transparent",
            "borderColor": "#F46152",
            "borderRadius": 9999,
            "borderStyle": "solid",
            "borderWidth": 2,
            "height": 48,
            "justifyContent": "center",
            "paddingLeft": 20,
            "paddingRight": 20,
            "width": "100%",
          },
        ]
      }
    >
      <Text
        disabled={false}
        inverted={false}
        pressed={false}
        style={
          Array [
            Object {
              "color": "#F46152",
              "fontSize": 14,
              "fontWeight": "700",
              "textAlign": "center",
            },
          ]
        }
      >
        Gympass
      </Text>
    </View>
  </TouchableWithoutFeedback>
</View>
`;

exports[`<Button /> Snapshots With full prop should match snapshot with text Button 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableWithoutFeedback
    full={true}
  >
    <View
      disabled={false}
      full={true}
      pressed={false}
      small={false}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "backgroundColor": "transparent",
            "borderColor": "transparent",
            "borderRadius": 9999,
            "borderStyle": "solid",
            "borderWidth": 2,
            "height": 48,
            "justifyContent": "center",
            "paddingLeft": 20,
            "paddingRight": 20,
            "width": "100%",
          },
        ]
      }
    >
      <Text
        disabled={false}
        pressed={false}
        style={
          Array [
            Object {
              "color": "#F46152",
              "fontSize": 14,
              "fontWeight": "700",
              "textAlign": "center",
            },
          ]
        }
      >
        Gympass
      </Text>
    </View>
  </TouchableWithoutFeedback>
</View>
`;

exports[`<Button /> Snapshots With inverted prop should match snapshot with default Button 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableWithoutFeedback
    inverted={true}
  >
    <View
      disabled={false}
      full={false}
      inverted={true}
      pressed={false}
      small={false}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "backgroundColor": "#FFFFFF",
            "borderColor": "#FFFFFF",
            "borderRadius": 9999,
            "borderStyle": "solid",
            "borderWidth": 2,
            "height": 48,
            "justifyContent": "center",
            "paddingLeft": 20,
            "paddingRight": 20,
          },
        ]
      }
    >
      <Text
        disabled={false}
        inverted={true}
        pressed={false}
        style={
          Array [
            Object {
              "color": "#F46152",
              "fontSize": 14,
              "fontWeight": "700",
              "textAlign": "center",
            },
          ]
        }
      >
        Gympass
      </Text>
    </View>
  </TouchableWithoutFeedback>
</View>
`;

exports[`<Button /> Snapshots With inverted prop should match snapshot with outline Button 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableWithoutFeedback
    inverted={true}
  >
    <View
      disabled={false}
      full={false}
      inverted={true}
      pressed={false}
      small={false}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "backgroundColor": "transparent",
            "borderColor": "#FFFFFF",
            "borderRadius": 9999,
            "borderStyle": "solid",
            "borderWidth": 2,
            "height": 48,
            "justifyContent": "center",
            "paddingLeft": 20,
            "paddingRight": 20,
          },
        ]
      }
    >
      <Text
        disabled={false}
        inverted={true}
        pressed={false}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 14,
              "fontWeight": "700",
              "textAlign": "center",
            },
          ]
        }
      >
        Gympass
      </Text>
    </View>
  </TouchableWithoutFeedback>
</View>
`;

exports[`<Button /> Snapshots With inverted prop should match snapshot with text Button 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableWithoutFeedback
    inverted={true}
  >
    <View
      disabled={false}
      full={false}
      inverted={true}
      pressed={false}
      small={false}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "backgroundColor": "transparent",
            "borderColor": "transparent",
            "borderRadius": 9999,
            "borderStyle": "solid",
            "borderWidth": 2,
            "height": 48,
            "justifyContent": "center",
            "paddingLeft": 20,
            "paddingRight": 20,
          },
        ]
      }
    >
      <Text
        disabled={false}
        pressed={false}
        style={
          Array [
            Object {
              "color": "#F46152",
              "fontSize": 14,
              "fontWeight": "700",
              "textAlign": "center",
            },
          ]
        }
      >
        Gympass
      </Text>
    </View>
  </TouchableWithoutFeedback>
</View>
`;

exports[`<Button /> Snapshots With small prop should match snapshot with default Button 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableWithoutFeedback
    small={true}
  >
    <View
      disabled={false}
      full={false}
      inverted={false}
      pressed={false}
      small={true}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "backgroundColor": "#F46152",
            "borderColor": "#F46152",
            "borderRadius": 9999,
            "borderStyle": "solid",
            "borderWidth": 1,
            "height": 32,
            "justifyContent": "center",
            "paddingLeft": 20,
            "paddingRight": 20,
          },
        ]
      }
    >
      <Text
        disabled={false}
        inverted={false}
        pressed={false}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 14,
              "fontWeight": "700",
              "textAlign": "center",
            },
          ]
        }
      >
        Gympass
      </Text>
    </View>
  </TouchableWithoutFeedback>
</View>
`;

exports[`<Button /> Snapshots With small prop should match snapshot with outline Button 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableWithoutFeedback
    small={true}
  >
    <View
      disabled={false}
      full={false}
      inverted={false}
      pressed={false}
      small={true}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "backgroundColor": "transparent",
            "borderColor": "#F46152",
            "borderRadius": 9999,
            "borderStyle": "solid",
            "borderWidth": 1,
            "height": 32,
            "justifyContent": "center",
            "paddingLeft": 20,
            "paddingRight": 20,
          },
        ]
      }
    >
      <Text
        disabled={false}
        inverted={false}
        pressed={false}
        style={
          Array [
            Object {
              "color": "#F46152",
              "fontSize": 14,
              "fontWeight": "700",
              "textAlign": "center",
            },
          ]
        }
      >
        Gympass
      </Text>
    </View>
  </TouchableWithoutFeedback>
</View>
`;

exports[`<Button /> Snapshots With small prop should match snapshot with text Button 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableWithoutFeedback
    small={true}
  >
    <View
      disabled={false}
      full={false}
      pressed={false}
      small={true}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "backgroundColor": "transparent",
            "borderColor": "transparent",
            "borderRadius": 9999,
            "borderStyle": "solid",
            "borderWidth": 1,
            "height": 32,
            "justifyContent": "center",
            "paddingLeft": 20,
            "paddingRight": 20,
          },
        ]
      }
    >
      <Text
        disabled={false}
        pressed={false}
        style={
          Array [
            Object {
              "color": "#F46152",
              "fontSize": 14,
              "fontWeight": "700",
              "textAlign": "center",
            },
          ]
        }
      >
        Gympass
      </Text>
    </View>
  </TouchableWithoutFeedback>
</View>
`;

exports[`<Button /> Snapshots Without props should match snapshot with default Button 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableWithoutFeedback>
    <View
      disabled={false}
      full={false}
      inverted={false}
      pressed={false}
      small={false}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "backgroundColor": "#F46152",
            "borderColor": "#F46152",
            "borderRadius": 9999,
            "borderStyle": "solid",
            "borderWidth": 2,
            "height": 48,
            "justifyContent": "center",
            "paddingLeft": 20,
            "paddingRight": 20,
          },
        ]
      }
    >
      <Text
        disabled={false}
        inverted={false}
        pressed={false}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 14,
              "fontWeight": "700",
              "textAlign": "center",
            },
          ]
        }
      >
        Gympass
      </Text>
    </View>
  </TouchableWithoutFeedback>
</View>
`;

exports[`<Button /> Snapshots Without props should match snapshot with outline Button 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableWithoutFeedback>
    <View
      disabled={false}
      full={false}
      inverted={false}
      pressed={false}
      small={false}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "backgroundColor": "transparent",
            "borderColor": "#F46152",
            "borderRadius": 9999,
            "borderStyle": "solid",
            "borderWidth": 2,
            "height": 48,
            "justifyContent": "center",
            "paddingLeft": 20,
            "paddingRight": 20,
          },
        ]
      }
    >
      <Text
        disabled={false}
        inverted={false}
        pressed={false}
        style={
          Array [
            Object {
              "color": "#F46152",
              "fontSize": 14,
              "fontWeight": "700",
              "textAlign": "center",
            },
          ]
        }
      >
        Gympass
      </Text>
    </View>
  </TouchableWithoutFeedback>
</View>
`;

exports[`<Button /> Snapshots Without props should match snapshot with text Button 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableWithoutFeedback>
    <View
      disabled={false}
      full={false}
      pressed={false}
      small={false}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "backgroundColor": "transparent",
            "borderColor": "transparent",
            "borderRadius": 9999,
            "borderStyle": "solid",
            "borderWidth": 2,
            "height": 48,
            "justifyContent": "center",
            "paddingLeft": 20,
            "paddingRight": 20,
          },
        ]
      }
    >
      <Text
        disabled={false}
        pressed={false}
        style={
          Array [
            Object {
              "color": "#F46152",
              "fontSize": 14,
              "fontWeight": "700",
              "textAlign": "center",
            },
          ]
        }
      >
        Gympass
      </Text>
    </View>
  </TouchableWithoutFeedback>
</View>
`;
